File: webtester/nodejs/features/support/world.js

require('chromedriver')
const seleniumWebdriver = require('selenium-webdriver');
var utils = require('..\\support\\utils');
const {defineSupportCode} = require('cucumber');

function CustomWorld() {
    this.driver = new seleniumWebdriver.Builder()
        .forBrowser('chrome')
        .build();

    this.loginCredentials = new Map();

    this.getLoginEmail = function() {

        if(this.loginCredentials.has("email")) {
            return this.loginCredentials.get("email");
        }
        else {
            const email = utils.generateTestEmail();
            this.loginCredentials.set("email", email);
            return email;
        }
    };

    this.getLoginPassword = function() {

        if(this.loginCredentials.has("password")) {
            return this.loginCredentials.get("password");
        }
        else {
            const password = utils.generatePassword();
            this.loginCredentials.set("password", password);
            return password;
        }
    };

}

defineSupportCode(function({setWorldConstructor}) {
    setWorldConstructor(CustomWorld)
})

File: webtester/java/src/test/java/com/mytesting/World.java
package com.mytesting;

import java.util.HashMap;
import cucumber.runtime.java.guice.ScenarioScoped;

@ScenarioScoped
public class World {

	private HashMap<String,String> loginCredentials = new HashMap<>();

	public String getLoginEmail() {
		
		if(loginCredentials.containsKey("email")) {
			return loginCredentials.get("email");
		}
		
		String email = Utils.generateTestEmail();
		loginCredentials.put("email", email);
		return email;
	}
	
	public String getLoginPassword() {
			
		if(loginCredentials.containsKey("password")) {
			return loginCredentials.get("password");
		}
		
		String pwd = Utils.generateTestPassword(8);
		loginCredentials.put("password", pwd);
		return pwd;
	}

    
}
